backpack/core/derivatives/basederivatives.py

```diff
@@ -124,17 +128,36 @@ def ea_jac_t_mat_jac_prod(self, module, g_inp, g_out, mat):
         """
         raise NotImplementedError
 
-    def hessian_is_zero(self):
+    def hessian_is_zero(self) -> bool:
+        """Whether ``∂²output[i] / ∂input[j] ∂input[k] = 0  ∀ i,j,k``."""
         raise NotImplementedError
 
-    def hessian_is_diagonal(self):
-        """Is `∂²output[i] / ∂input[j] ∂input[k]` nonzero only if `i = j = k`."""
+    def hessian_is_diagonal(self) -> bool:
+        """Is `∂²output[i] / ∂input[j] ∂input[k]` nonzero only if `i = j = k`.
+
+        The Hessian diagonal is only defined for layers that preserve the size
+        of their input.
+
+        Must be implemented by descendants that don't implement ``hessian_is_zero``.
+        """
         raise NotImplementedError
```

Comment:
Remove from interface, replace calls to `hessian_is_diagonal()` by `issubclass(derivative, ElementwiseDerivatives)`